drizzle-zod/src/mysql/index.ts

```diff
@@ -0,0 +1,243 @@
+import { GetColumnConfig } from 'drizzle-orm';
+import {
+	AnyPgColumn,
+	AnyPgTable,
+	GetTableConfig,
+	PgBigInt53,
+	PgBigInt64,
+	PgBigSerial53,
+	PgBigSerial64,
+	PgBoolean,
+	PgDate,
+	PgDoublePrecision,
+	PgEnumColumn,
+	PgEnumColumnConfig,
+	PgInteger,
+	PgInterval,
+	PgNumeric,
+	PgReal,
+	PgSerial,
+	PgSmallInt,
+	PgSmallSerial,
+	PgText,
+	PgTextConfig,
+	PgTime,
+	PgTimestamp,
+	PgUUID,
+	PgVarchar,
+} from 'drizzle-orm/mysql-core';
+import { getTableColumns } from 'drizzle-orm/pg-core/utils';
+import { Simplify } from 'drizzle-orm/utils';
+import { z } from 'zod';
+
+type SnakeToCamelCase<S extends string> = S extends `${infer T}${'_' | '-'}${infer U}`
+	? `${T}${Capitalize<SnakeToCamelCase<U>>}`
+	: S;
+
+type ToSnakeCase<S extends string> = S extends `${infer T}${'' | '-'}${infer U}`
+	? `${T extends Capitalize<T> ? Lowercase<T> extends Capitalize<T> ? '' : '_' : ''}${Lowercase<T>}${ToSnakeCase<U>}`
+	: S;
+
+function toCamelCase(value: string): string {
+	return value.replace(/([-_][a-z])/gi, ($1) => $1.toUpperCase().replace('-', '').replace('_', ''));
+}
+
+function toSnakeCase(value: string): string {
+	return value.replace(/([A-Z])/g, ($1) => `_${$1.toLowerCase()}`);
+}
+
+type MaybeOptional<TColumn extends AnyPgColumn, TType extends z.ZodTypeAny, TNoOptional extends boolean = false> =
+	TNoOptional extends true ? TType
+		: GetColumnConfig<TColumn, 'hasDefault'> extends true ? z.ZodOptional<TType>
+		: TType;
+
+type GetZodType<TColumn extends AnyPgColumn> = GetColumnConfig<TColumn, 'data'> extends infer TType
+	? TColumn extends PgEnumColumn<PgEnumColumnConfig & { values: infer TValues extends [string, ...string[]] }>
+		? z.ZodEnum<TValues>
+	: TType extends number ? z.ZodNumber
+	: TType extends string ? z.ZodString
+	: TType extends boolean ? z.ZodBoolean
+	: TType extends Date ? z.ZodDate
+	: z.ZodAny
+	: never;
+
+type ConvertKeyName<K extends string, TCase extends 'snake' | 'camel' | undefined> = undefined extends TCase ? K
+	: 'snake' extends TCase ? ToSnakeCase<K>
+	: SnakeToCamelCase<K>;
+
+export type BuildInsertSchema<
+	TTable extends AnyPgTable,
```

Comment:
`AnyPgTable` is used for postgres specific functions. We need to use `AnyMySqlTable` here